File: backend/src/main/java/es/udc/tfg/backend/model/services/RoomTypeService.java
package es.udc.tfg.backend.model.services;

import java.util.List;

import es.udc.tfg.backend.model.common.exceptions.DuplicateInstanceException;
import es.udc.tfg.backend.model.common.exceptions.InstanceNotFoundException;
import es.udc.tfg.backend.model.entities.RoomType;

public interface RoomTypeService {

	RoomType addRoomType(Long hotelId, RoomType roomType) throws DuplicateInstanceException, InstanceNotFoundException;
	
	RoomType findRoomTypeById(Long id) throws InstanceNotFoundException;

	RoomType updateRoomType(Long hotelId, RoomType roomType) throws InstanceNotFoundException, PermissionException;

	void removeRoomType(Long hotelId, Long roomTypeId) throws InstanceNotFoundException, PermissionException;

	List<RoomType> findAllRoomTypes();

}

File: frontend/src/backend/roomTypeService.js
 
import {config, appFetch } from './appFetch';


export const addRoomType = (roomType, onSuccess, onErrors) => {
    let path = `/roomTypes/addRoomType`;
    console.log(path)
    appFetch(path, config('POST',  roomType), 
    onSuccess, onErrors)};

export const findAllRoomTypes = (onSuccess,onErrors) => 
    appFetch('/roomTypes/roomTypes', config('GET'),
    onSuccess, onErrors);

 export const updateRoomType = (roomType, onSuccess, onErrors) =>
    appFetch(`/roomTypes/${roomType.id}`, config('PUT', roomType),
    onSuccess, onErrors);

export const removeRoomType = (roomType, onSuccess, onErrors) =>{
    appFetch(`/roomTypes/${roomType.id}`, config('DELETE'), 
    onSuccess, onErrors);}

export const findRoomTypeById = (id, onSuccess, onErrors) =>{
    appFetch(`/roomTypes/${id}`, config('GET'), 
    onSuccess, onErrors);}